apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -1169,13 +1169,21 @@ func (m *ModelManagerAPI) getModelInfo(ctx context.Context, tag names.ModelTag,
 		info.AgentVersion = &agentVersion
 	}
 
-	status, err := model.Status()
+	status, err := modelInfoService.GetStatus(ctx)
 	if shouldErr(err) {
 		return params.ModelInfo{}, errors.Trace(err)
 	}
-	if err == nil {
```

Comment:
We still need this check right? shouldErr(err) will return false for some types of error, meaning err can still be non nil after this call and status will not be valid.

Reply:
If we don't have tests for cases where `shouldErr` returns false but err is != nil we should add them.

Reply:
Good pickup. Have fixed.

We aren't doing tests as part of the agreed plan for this task. This API will be completely refactor in a few pulses time when the core logic moves down into services.